### ISSUE TITLE
Prevent use of older clients with already-upgraded databases

### ISSUE THREAD

**Issue by fmoc** [owncloud/client#10964]
Follow-up from https://github.com/owncloud/client/issues/10473. Needs to be worked on once the former one is merged.

https://github.com/owncloud/client/blob/cde9c61e5aa5c2e1661ea5a6d225b2e1752d487b/src/common/syncjournaldb.cpp#L523-L523

**Comment by fmoc** [comment]
A new version check is required after the config file check mentioned in #10473. We may be able to use the same dialog if we implement this check in between.

**Comment by erikjv** [comment]
To do:

1. add a DB version check in `FolderMan::setupFoldersHelper`, and make it fail-able. 
2. move `FolderMan::instance()->setupFolders();` out of the `Application` constructor, because it can fail (together with showing the wizard if no accounts exist)

For 1. above: do this right after the call to `SyncJournalDb::maybeMigrateDb`? Maybe create a temporary instance of `SyncJournalDb` just to do a version check?

**Comment by erikjv** [comment]
@TheOneRing ^

**Comment by TheOneRing** [comment]
> To do:
> 
>     1. add a DB version check in `FolderMan::setupFoldersHelper`, and make it fail-able.
> 
>     2. move `FolderMan::instance()->setupFolders();` out of the `Application` constructor, because it can fail (together with showing the wizard if no accounts exist)
> 
> 
> For 1. above: do this right after the call to `SyncJournalDb::maybeMigrateDb`? Maybe create a temporary instance of 

Sounds good, `SyncJournalDb::versionCheck` (We are again not migrating).

>`SyncJournalDb` just to do a version check?

**Comment by Salipa-Gurung** [comment]
How can I confirm that it prevents use of older clients with already-upgraded databases?

**Comment by erikjv** [comment]
Go to a sync root, then on the command line:

```
sqlite3 .sync_journal.db
sqlite> select major from version;
5
sqlite> update version set major=6 where major=5;
sqlite> select major from version;
6
```

**Comment by Salipa-Gurung** [comment]
### Desktop Client Version
ownCloud 5.0.0.11804-daily20230818 [65017f](https://github.com/owncloud/client/commit/65017f978ed45f3f2ff0e639c367bb63d6e7e333)
ubuntu-6.2.0-26-generic

#### With upgraded database user is prevented from using older client version :heavy_check_mark: 

![Screenshot from 2023-08-18 12-01-24](https://github.com/owncloud/client/assets/53458341/7ca8cbc5-0c22-4eb4-acf2-334a277a7034) | ![Screenshot from 2023-08-18 12-04-14](https://github.com/owncloud/client/assets/53458341/dd8683a0-d442-4dbf-9c8b-5a269225c8ca)
---|---

**Comment by saw-jan** [comment]
^
was crash dialog expected in this case?
CC @TheOneRing @erikjv

**Comment by TheOneRing** [comment]
Crash was not expected, please provide a crash report.

**Comment by Salipa-Gurung** [comment]
Crash id : `2dbe43a9-4130-43ae-b62e-1f0a58f235cf`
Logs [ownCloud.log](https://github.com/owncloud/client/files/12378217/ownCloud.log)

**Comment by fmoc** [comment]
Fixed in #11137, please re-test.

**Comment by Salipa-Gurung** [comment]
#### Desktop client version : `ownCloud 5.0.0.11938-daily20230828 84a468`
- Older client version cannot be used with upgraded database :heavy_check_mark: 
- Crash dialog does not appear :heavy_check_mark: 

Server : Infinite Scale 3.0.0 Community 
ubuntu-6.2.0-26-generic | windows-10.0.19045
---|---
![Screenshot from 2023-08-28 10-02-00](https://github.com/owncloud/client/assets/53458341/1c683883-c0e5-4cda-a597-ce83b3932c90) | ![Screenshot from 2023-08-28 10-04-57](https://github.com/owncloud/client/assets/53458341/d3601082-ce7d-4ec3-a642-30b653103780)


Server: ownCloud 10.13.0 (stable)
ubuntu-6.2.0-26-generic | windows-10.0.19045
---|---
![Screenshot from 2023-08-28 09-46-32](https://github.com/owncloud/client/assets/53458341/97c359dc-1156-4bb9-b677-1610630f6185) | ![Screenshot from 2023-08-28 09-54-25](https://github.com/owncloud/client/assets/53458341/de4e76be-6346-4ead-adf7-56adcf5d953e)
